File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxArea = 0;
        int left = 0;
        int right = height.length - 1;  // Initialize 'right' at the end of the array

        // Loop until the two pointers meet
        while (left < right) {
            // Calculate the current area
            int currentArea = (right - left) * Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, currentArea);

            // Move the pointer pointing to the shorter line
            if (height[left] < height[right]) {
                left++;  // Move left pointer right
            } else {
                right--;  // Move right pointer left
            }
        }

        return maxArea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = 0
        left = 0
        right = len(height) - 1  # Initialize the right pointer at the end

        # Move pointers towards each other
        while left < right:
            # Calculate the current area using the shorter height
            current_area = (right - left) * min(height[left], height[right])
            # Update the maximum area found so far
            max_area = max(max_area, current_area)

            # Move the pointer corresponding to the shorter line
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        
        return max_area
